cmf-cli/Commands/new/iot/GenerateDotnetDriverCommand.cs

```diff
@@ -0,0 +1,130 @@
+using Cmf.CLI.Core.Enums;
+using Cmf.CLI.Core.Objects;
+using Cmf.CLI.Utilities;
+using Microsoft.Extensions.DependencyInjection;
+using Spectre.Console;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.NamingConventionBinder;
+using System.IO.Abstractions;
+
+namespace Cmf.CLI.Commands.New.IoT
+{
+    /// <summary>
+    /// Generates IoT Dotnet Driver structure
+    /// </summary>
+    [CmfCommand("driver-dotnet", ParentId = "new_iot", Id = "iot_dotnet_driver")]
+    public class GenerateDotnetDriverCommand : TemplateCommand
+    {
+        /// <summary>
+        /// constructor
+        /// </summary>
+        public GenerateDotnetDriverCommand() : base("driver-dotnet")
+        {
+        }
+
+        /// <summary>
+        /// constructor
+        /// </summary>
+        /// <param name="fileSystem">the filesystem implementation</param>
+        public GenerateDotnetDriverCommand(IFileSystem fileSystem) : base("driver-dotnet", fileSystem)
+        {
+        }
+
+        public override void Configure(Command cmd)
+        {
+            var nearestIoTPackage = FileSystemUtilities.GetPackageRootByType(
+                this.fileSystem.Directory.GetCurrentDirectory(),
+                PackageType.IoT,
+                this.fileSystem
+            );
+
+            cmd.AddArgument(new Argument<IDirectoryInfo>(
+                name: "workingDir",
+                parse: (argResult) => Parse<IDirectoryInfo>(argResult, nearestIoTPackage?.FullName),
+                isDefault: true
+            )
+            {
+                Description = "Working Directory"
+            });
+
+            cmd.Handler = CommandHandler.Create<IDirectoryInfo>(this.Execute);
+        }
+
+        /// <summary>
+        /// Execute the command
+        /// </summary>
+        /// <param name="workingDir">nearest root package</param>
+        /// <param name="version">package version</param>
+        /// <param name="htmlPackageLocation">location of html package</param>
+        public void Execute(IDirectoryInfo workingDir)
+        {
+            if (workingDir == null)
+            {
+                throw new CliException("This command needs to run inside an iot project. Run `cmf new iot` to create a new project.");
+            }
+
+            if (ExecutionContext.Instance.ProjectConfig.MESVersion.Major < 11)
+            {
+                throw new CliException("This command is only valid for versions above 11.0.0");
+            }
+
+            using var activity = ExecutionContext.ServiceProvider?.GetService<ITelemetryService>()?.StartExtendedActivity(this.GetType().Name);
+
+            var driver = HandleDriver(new DriverValues());
+
+            var args = this.GenerateArgs(workingDir, this.fileSystem.Directory.GetCurrentDirectory(), driver.Directory, driver.Identifier, driver.IdentifierCamel, driver.PackageFullName, driver.PackageVersion, driver.HasCommands);
+            this.CommandName = "iot-driver-dotnet";
```

Comment:
This shouldn't be needed, the does the template have a different name from the command?

Reply:
even so, the template name should be specific when calling the base constructor, at lines 25 and 33. We only override when multiple templates can be called in the same command.